magento/magento2
# Fixed 'Undefined class constant' error when interceptor is generated.

### Description (*)
1. Since constant is [declared as private](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Theme/Block/Html/Title.php#L27) it's not accessible from parent classes.
2. Since it used [with `static` keyword](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Theme/Block/Html/Title.php#L104), PHP will search it within class instance.
3. Class instance will be an instance of generated `Interceptor` without access to private variables of child classes.
4. This will lead to an 'Undefined class constant' error.

### Manual testing scenarios (*)
1. Create a plugin for `Magento\Theme\Block\Html\Title` class:

    ```xml
    <type name="Magento\Theme\Block\Html\Title">
        <plugin name="titlePlugin" type="Magento\Theme\Plugin\TitlePlugin" />
    </type>
    ```

    ```php
    <?php

    namespace Magento\Theme\Plugin;

    class TitlePlugin
    {
        /**
         * @param mixed $subject
         * @param mixed $result
         * @return mixed
         */
        public function afterGetPageHeading(
            $subject,
            $result
        ) {
            return $result;
        }
    }
    ```

2. Open frontend.
3. You'll see an error:

    ```
    Error: Undefined class constant 'XML_PATH_HEADER_TRANSLATE_TITLE' in app/code/Magento/Theme/Block/Html/Title.php:104 
    Stack trace: 
    #0 app/code/Magento/Theme/Block/Html/Title.php(82): Magento\Theme\Block\Html\Title->shouldTranslateTitle() 
    #1 generated/code/Magento/Theme/Block/Html/Title/Interceptor.php(32): Magento\Theme\Block\Html\Title->getPageHeading() 
    #2 app/code/Magento/Theme/view/frontend/templates/html/title.phtml(12): Magento\Theme\Block\Html\Title\Interceptor->getPageHeading()
    ...
    ```

### Questions or comments
Another way to solve this bug - is to declare constant as public instead of private.
If you think it should be public - I'll update PR.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#28981: Fixed 'Undefined class constant' error when interceptor is generated.
